searvey/erddap.py

```diff
@@ -48,7 +48,7 @@ def get_erddap_url(
     response: str = "csvp",
 ) -> str:
     erddap = erddapy.ERDDAP(
-        server=dataset.server_url,
+        server=str(dataset.server_url),
```

Comment:
I haven't used ERDDAP module, so I'm not sure if this is relevant. But is it possible for the value to become `None` and making it `str()` can break some check downstream?

Reply:
Well the ERDDAP stuff hasn't been used/tested in some time. If we decide to add a Provider using an ERDDAP server we will need to revisit it anyhow. For now I just made the change so that the tests pass.

This specific change is actually caused by pydantic 2+ which changed some APIs.